utils/pvacapi/controllers/database.py

```diff
@@ -211,22 +264,26 @@ def create_table(parentID, fileID, data, tablekey, db):
                         )
                     print("Alter:", update + ','.join(alter_cols))
                     db.execute(update + ','.join(alter_cols))
-                #TEMPORARY: remove "protein_position" column from tables as visualization of ranges not currently possible
+                #remove "protein_position" column from tables as visualization of ranges not currently possible
                 db.execute(update + "DROP COLUMN protein_position")
         except psql.exceptions.UniqueError: #If another transaction already created specified table, pass
             pass
-        except psql.exceptions.InsufficientPrivilegeError:
+        except psql.exceptions.InsufficientPrivilegeError as e:
             #can occur when postgres user unable to open file due to permissions; specifically for travis-ci tests
-            #attempt to resolve by copying file to /tmp/, changing its permissions, and accessing it there
-            if file_permissions:
-                print("InsufficientPrivilegeError raised in accessing file.\nAttempting workaround...")
-                file_permissions = False
+            #check if resulting from postgres user permissions
+            if e.args[0].startswith("must be superuser"):
+                print("WARNING: Postgres user is not a super user; visualization time may be slow")
+                old_file_read(db, CREATE_TABLE, tablekey, column_names, reader, mapping) #use inefficient file-read-to-db method
             else:
-                print("Postgres could not access file.  Check to make sure that both the "
-                    "file and your current postgres user has the appropriate permissions.")
-                raise
-        else:
-            retry_transaction = False
+                #attempt to resolve by copying file to /tmp/, changing its permissions, and accessing it there
+                if file_permissions:
+                    print("InsufficientPrivilegeError raised in accessing file.\nAttempting workaround...")
+                    retry_transaction = True
```

Comment:
If you put everything from line 212-268 in its own method (making `file_permissions` a parameter), you can simply call that subroutine here with file_permission set to `False`. This would remove the whole `while` loop and the `retry_transaction` flag.